emu/src/arm7tdmi.rs

```diff
@@ -82,14 +82,12 @@ impl Cpu for Arm7tdmi {
         // Instruction functions should return whether PC has to be advanced
         // after instruction executed.
         let should_advance_pc = match op_code.instruction {
-            Branch => self.branch(op_code, false),
-            BranchLink => self.branch(op_code, true),
             DataProcessing1 | DataProcessing2 | DataProcessing3 => self.data_processing(op_code),
+            BranchAndBranchWithLink => self.branch(op_code, true),
+            BranchAndExchange => self.branch(op_code, false),
```

Comment:
I'd put `unimplemented!` here since it's `BX` is different than a regular `branch`